R/qualification-comparison-time-profile.R

```diff
@@ -136,90 +136,90 @@ addOutputToComparisonTimeProfile <- function(outputMapping, simulationDuration,
     plotObject = plotObject
   )
 
-  # Get observed output
-  observedResults <- getObservedDataFromConfigurationPlan(outputMapping$ObservedData, configurationPlan, logFolder)
-  observedTime <- ospsuite::toUnit(
-    quantityOrDimension = "Time",
-    values = as.numeric(observedResults$data[, 1]),
-    targetUnit = axesProperties$x$unit,
-    sourceUnit = observedResults$metaData$time$unit
-  )
-  observedTime <- observedTime - timeOffset
-  selectedObservedTimeValues <- observedTime >= 0 & observedTime <= simulationDuration
-  observedTime <- observedTime[selectedObservedTimeValues]
-
-  logWorkflow(
-    paste0(
-      "In comparison time profile, '", sum(selectedObservedTimeValues), "' values selected between ",
-      timeOffset, " and ", timeOffset + simulationDuration, " ", axesProperties$x$unit,
-      " in Observed Dataset '", outputMapping$ObservedData, "'"
-    ),
-    logFolder,
-    LogTypes$Debug
-  )
-
-  observedValues <- ospsuite::toUnit(
-    quantityOrDimension = ospsuite::getDimensionForUnit(observedResults$metaData$output$unit),
-    values = observedResults$data[, 2],
-    targetUnit = axesProperties$y$unit,
-    sourceUnit = tolower(observedResults$metaData$output$unit),
-    molWeight = molWeight
-  )
-  observedValues <- observedValues[selectedObservedTimeValues]
-  observedError <- NULL
-  if (!ospsuite.utils::isOfLength(observedResults$metaData$error, 0)) {
-    # No unit means that error is geometric
-    observedError$ymin <- observedValues / observedResults$data[, 3]
-    observedError$ymax <- observedValues * observedResults$data[, 3]
-
-    if (!ospsuite.utils::isIncluded(observedResults$metaData$error$unit, "")) {
-      observedError$ymin <- observedValues - ospsuite::toUnit(
-        ospsuite::getDimensionForUnit(observedResults$metaData$error$unit),
-        observedResults$data[, 3],
-        targetUnit = axesProperties$y$unit,
-        sourceUnit = observedResults$metaData$error$unit,
-        molWeight = molWeight
-      )
-
-      observedError$ymax <- observedValues + ospsuite::toUnit(
-        ospsuite::getDimensionForUnit(observedResults$metaData$error$unit),
-        observedResults$data[, 3],
-        targetUnit = axesProperties$y$unit,
-        sourceUnit = observedResults$metaData$error$unit,
-        molWeight = molWeight
+  # Loop on each observed dataset in OutputMappings
+  for (observedDataSet in outputMapping$ObservedData) {

```

Comment:
@pchelle this would work with both string **value** and string **array** in the json configuration plan?

Reply:
Yes.
I have tested the example, it works with both.
It also worked when I tested the addition of a second observed data in the array